src/server/controlchan/commands/pass.rs

```diff
@@ -105,10 +105,34 @@ where
                             } else if user.account_enabled() {
                                 let mut session = session2clone.lock().await;
                                 slog::info!(logger, "PASS: User {} logged in", user);
-                                session.user = Arc::new(Some(user));
-                                ControlChanMsg::AuthSuccess {
-                                    username,
-                                    trace_id: session.trace_id,
+                                if let Some(home) = user.home() {
```

Comment:
```rust
} else if user.account_enabled() {
                                let mut session = session2clone.lock().await;
                                slog::info!(logger, "PASS: User {} logged in", user);
                                // Using Arc::get_mut means that this won't work if the Session
                                // is currently servicing multiple commands concurrently.  But
                                // it shouldn't ever be servicing PASS at the same time as
                                // another command.
                                match Arc::get_mut(&mut session.storage).map(|s| s.enter(&user)) {
                                    Some(Err(e)) => {
                                        slog::error!(logger, "{}", e);
                                        ControlChanMsg::AuthFailed
                                    }
                                    None => {
                                        slog::error!(logger, "Failed to lock Session::storage during PASS.");
                                        ControlChanMsg::AuthFailed
                                    }
                                    Some(Ok(())) => {
                                        session.user = Arc::new(Some(user));
                                        ControlChanMsg::AuthSuccess {
                                            username,
                                            trace_id: session.trace_id,
                                        }
                                    }
                                }
                            }
```

Reply:
when we adapt the `enter` method to take a `UserDetail` implementation instead of a path.